guides/release/routing/specifying-a-routes-model.md

```diff
@@ -161,13 +154,88 @@ each record in the song model and album model:
+## Dynamic Models
+
+In the examples above, we showed a route that will always return the same data, a collection of favorite posts. Even when the user leaves and re-enters the `/posts` route, they will see the same thing.
+But what if you need to request different data after user interaction?
+What if a specific post should load based on the URL that the user visited, like `posts/42`?
+In Ember, this can be accomplished by defining routes with [dynamic
+segments](../defining-your-routes/#toc_dynamic-segments), or by using [query parameters](/query-params), and then using the dynamic data to make requests.
+
+In the previous Guides topic, we showed making a dynamic segment in the app's `router.js`:
+
+```javascript {data-filename=app/router.js}
+Router.map(function() {
+  this.route('posts');
+  this.route('post', { path: '/post/:post_id' });
+});
+```
+
+Whatever shows up in the URL at the `:post_id`, the dynamic segment, will be available in the params for the route's `model` hook:
+
+```javascript {data-filename=app/routes/photo.js}
+import Route from '@ember/routing/route';
+
+export default Route.extend({
+  model(params) {
+    console.log('This is the dynamic segment data: ' + params.post_id)
+    // make an API request that uses the id
+  }
+});
+```
+
+### Linking to a dynamic segment
+
+There are two ways to link to a dynamic segment from an `.hbs` template using the
+[link-to](../../templates/links/)
+helper.
+Depending on which approach you use, it will affect whether that route's `model` hook is run.
+To learn how to link to a dynamic segment from within the JavaScript file, see the API documentation on
+[`transitionTo`](https://api.emberjs.com/ember/release/classes/RouterService/methods/transitionTo?anchor=transitionTo)
+instead.
+
+When you provide a string or number to the `link-to`, the dynamic segment's `model` hook will run when the app transitions to the new route.
+In this example, `photo.id` might have an id of `4`:
+
+```handlebars
+{{#each model as |photo|}}
+  {{#link-to "photo" photo.id}}
+    link text to display
+  {{/link-to}}
+{{/each}}
+```
+
+However, if you provide the entire model context, the model hook for that URL segment will _not_ be run.
+For this reason, many Ember developers choose to pass only ids to `{{link-to}}` so that the behavior is consistent.
+
+Here's what it looks like to pass the entire `photo` record:
+
+```handlebars
+{{#each model as |photo|}}
+  {{#link-to "photo" photo}}
+    link text to display
+  {{/link-to}}
+{{/each}}
+```
+
+If you decide to pass the entire model, be sure to cover this behavior in your acceptance tests.
+
+If a route you are trying to link to has multiple dynamic segments, like `/photos/4/comments/18`, be sure to specify all the necessary information for each segment:
```

Comment:
so this isn't 100% true 🤔 and to be honest I don't know how you would teach the alternative because it's quite a weird behaviour.

So first of all this is true if you're anywhere in the app **not** under the `photos` route. But if you're on `/photos/4` you can link to 

```handlebars
{{#link-to "photos.photo.comments.comment" 18}}
  link text to display
{{/link-to}}
```

and it will keep you within the first dynamic segment and use 18 for the second 🙈 kinda strange but c'est la vie 🤷‍♂

Reply:
In the interest of teaching the happy path in the Guides, I think we should keep it as-is.